log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationFactory.java

```diff
@@ -333,6 +335,71 @@ public Configuration getConfiguration(
         return getConfiguration(loggerContext, name, configLocation);
     }
 
+    /**
+     * {@return a {@link Configuration} created using provided configuration location {@link URI}s}
+     * If the provided list of {@code URI}s is null or empty, {@code getConfiguration(loggerContext, name, (URI) null)} will be returned.
```

Comment:
Should we explain more explicitly the behavior of an empty list?

```suggestion
     * <p>If the provided list of {@code URI}s is empty, the configuration factory
     * attempts to load an implementation-dependent set of default locations.
     * If no configuration can be found, a {@link ConfigurationException} is thrown.</p>
```

Reply:
Improved as requested in 6a8987f83.